sql/src/test/java/io/crate/execution/engine/indexing/IndexWriterProjectorUnitTest.java

```diff
@@ -122,8 +145,93 @@ public void testNullPKValue() throws Throwable {
-        testingBatchConsumer.getResult();
+        List<Object[]> result = testingBatchConsumer.getResult();
+        // zero affected rows as a NULL as a PK value will result in an error
+        // the error must bubble as other rows might already have been written
+        assertThat(result.get(0)[0], is(0L));
+    }
+
+    @Test
+    public void testReturnSummary() throws Throwable {
+        setNewClusterStateFor(createShardRouting());
+
+        InputCollectExpression sourceInput = new InputCollectExpression(0);
+        InputCollectExpression sourceUriInput = new InputCollectExpression(2);
+
+        TransportCreatePartitionsAction transportCreatePartitionsAction = mock(TransportCreatePartitionsAction.class);
+        IndexWriterProjector indexWriter = new IndexWriterProjector(
+            clusterService,
+            new NodeJobsCounter(),
+            scheduler,
+            executor,
+            TestingHelpers.getFunctions(),
+            Settings.EMPTY,
+            Settings.EMPTY,
+            transportCreatePartitionsAction,
+            (request, listener) -> {
+                ShardResponse shardResponse = new ShardResponse();
+                for (ShardRequest.Item item : request.items()) {
+                    shardResponse.add(item.location());
+                }
+                listener.onResponse(shardResponse);
+            },
+            IndexNameResolver.forTable(BULK_IMPORT_IDENT),
+            RAW_SOURCE_REFERENCE,
+            Collections.singletonList(ID_IDENT),
+            Collections.<Symbol>singletonList(new InputColumn(1)),
+            null,
+            null,
+            sourceInput,
+            sourceUriInput,
+            () -> null,
+            Collections.singletonList(sourceInput),
+            Collections.singletonList(sourceUriInput),
+            20,
+            null,
+            null,
+            false,
+            false,
+            UUID.randomUUID(),
+            UpsertResultCollectors.newSummaryCollector(clusterService.localNode()));
+
+        RowN row1 = new RowN(new Object[]{new BytesRef("{\"y\": \"foo\"}"), 1, new BytesRef("file://tmp/import1.json")});
+        RowN row2 = new RowN(new Object[]{new BytesRef("{\"y\": \"bar\"}"), 2, new BytesRef("file://tmp/import2.json")});
+        BatchIterator<Row> batchIterator = InMemoryBatchIterator.of(Arrays.asList(row1, row2), SENTINEL);
+        batchIterator = indexWriter.apply(batchIterator);
+
+        TestingRowConsumer testingBatchConsumer = new TestingRowConsumer();
+        testingBatchConsumer.accept(batchIterator, null);
+
+        List<Object[]> result = testingBatchConsumer.getResult();
+        assertThat(printedTable(result.toArray(new Object[0][])), is(
+            "{id=[6e 31], name=[6e 6f 64 65 2d 6e 61 6d 65]}| file://tmp/import2.json| 1| 0| {}\n" +
+            "{id=[6e 31], name=[6e 6f 64 65 2d 6e 61 6d 65]}| file://tmp/import1.json| 1| 0| {}\n"));
```

Comment:
This test is flaky because the order of the result row entries is not ensured.
As integration tests also exists now, I'm in favour of removing this unit test at all.

Reply:
You could safe the two result strings in variables and then check for either `string1 + string2` or `string2 + string1`.

Reply:
yes sure, I just didn't want to put effort here as I'd remove it completely. this was more a left over from implementation iterations. so you'd like to keep it?